src/util/codec/mysql/json.rs

```diff
@@ -31,66 +35,144 @@ const TYPE_CODE_I64: u8 = 0x09;
 const TYPE_CODE_DOUBLE: u8 = 0x0b;
 const TYPE_CODE_STRING: u8 = 0x0c;
 
+const PRECEDENCE_BLOB: i32 = -1;
+const PRECEDENCE_BIT: i32 = -2;
+const PRECEDENCE_OPAQUE: i32 = -3;
+const PRECEDENCE_DATETIME: i32 = -4;
+const PRECEDENCE_TIME: i32 = -5;
+const PRECEDENCE_DATE: i32 = -6;
+const PRECEDENCE_BOOLEAN: i32 = -7;
+const PRECEDENCE_ARRAY: i32 = -8;
+const PRECEDENCE_OBJECT: i32 = -9;
+const PRECEDENCE_STRING: i32 = -10;
+const PRECEDENCE_NUMBER: i32 = -11;
```

Comment:
This would caused the following warning:
```
 this `match` has identical arm bodies
   --> src/util/codec/mysql/json.rs:118:32
    |
118 |             Json::Double(_) => PRECEDENCE_DOUBLE,
    |                                ^^^^^^^^^^^^^^^^^
    |
    = note: #[warn(match_same_arms)] on by default
note: same as this
   --> src/util/codec/mysql/json.rs:117:29
    |
117 |             Json::I64(_) => PRECEDENCE_INTEGER,
    |                             ^^^^^^^^^^^^^^^^^
note: consider refactoring into `Json::I64(_) | Json::Double(_)`
```
@hhkbp2